[bazel_ros2_rules] Unable to configure with `ros-humble-rosbag2-storage-mcap`

@kunimatsu-tri ran into this (Anzu issue 11114).

I can reproduce it in the following commit: c626d43
```
$ cd ros2_example_bazel_installed
$ bazel fetch ...
...
--- captured stderr ---
CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rosbag2_storage_mcap" links to target
  "mcap_vendor::mcap" but the target was not found.  Perhaps a find_package()
  call is missing for an IMPORTED target, or an ALIAS target is missing?


CMake Error in CMakeLists.txt:
  Imported target "rosbag2_storage_mcap::rosbag2_storage_mcap" includes
  non-existent path

    "/opt/ros/humble/include/rosbag2_storage_mcap"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

\cc @adityapande-1995 @calderpg-tri @IanTheEngineer

--- a/bazel_ros2_rules/ros2/resources/templates/ament_cmake_CMakeLists.txt.in
+++ b/bazel_ros2_rules/ros2/resources/templates/ament_cmake_CMakeLists.txt.in
@@ -42,6 +42,12 @@ elseif("@PACKAGE@" STREQUAL "console_bridge_vendor")
   # packages?
   find_package(console_bridge REQUIRED)
   target_link_libraries(${PROJECT_NAME} console_bridge::console_bridge)
+elseif("@PACKAGE@" STREQUAL "rosbag2_storage_mcap")
+  # Workaround https://github.com/RobotLocomotion/drake-ros/issues/315
+  find_package(mcap_vendor REQUIRED)
+  set_target_properties(rosbag2_storage_mcap::rosbag2_storage_mcap PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES ""
+  )
 else()
   ament_target_dependencies(${PROJECT_NAME} @PACKAGE@)
   # TODO(hidmic): figure out why this is sometimes necessary

